monai/transforms/croppad/array.py

```diff
@@ -414,6 +415,58 @@ def __call__(self, img: np.ndarray):
         return cropped
 
 
+class RandWeightedCrop(Randomizable, Transform):
+    """
+    Samples a list of `num_samples` image patches according to the provided `weight_map`.
+
+    Args:
+        spatial_size: the spatial size of the image patch e.g. [224, 224, 128].
+            If its components have non-positive values, the corresponding size of `img` will be used.
+        num_samples: number of samples (image patches) to take in the returned list.
+        weight_map: weight map used to generate patch samples. The weights must be non-negative.
+            Each element denotes a sampling weight of the spatial location. 0 indicates no sampling.
+            It should be a single-channel array in shape, for example, `(1, spatial_dim_0, spatial_dim_1, ...)`.
+    """
+
+    def __init__(
+        self, spatial_size: Union[Sequence[int], int], num_samples: int = 1, weight_map: Optional[np.ndarray] = None
+    ):
+        self.spatial_size = ensure_tuple(spatial_size)
+        self.num_samples = int(num_samples)
+        self.weight_map = weight_map
+        self.centers: List[np.ndarray] = []
+
+    def randomize(self, weight_map: np.ndarray) -> None:
+        self.centers = weighted_patch_samples(
+            spatial_size=self.spatial_size, w=weight_map[0], n_samples=self.num_samples, r_state=self.R
+        )  # using only the first channel as weight map
+
+    def __call__(self, img: np.ndarray, weight_map: Optional[np.ndarray] = None) -> List[np.ndarray]:
+        """
+        Args:
+            img: input image to sample patches from. assuming `img` is a channel-first array.
+            weight_map: weight map used to generate patch samples. The weights must be non-negative.
+                Each element denotes a sampling weight of the spatial location. 0 indicates no sampling.
+                It should be a single-channel array in shape, for example, `(1, spatial_dim_0, spatial_dim_1, ...)`
+
+        Returns:
+            A list of image patches
+        """
+        if weight_map is None:
+            weight_map = self.weight_map
+        if weight_map is None:
+            raise ValueError("weight map must be provided for weighted patch sampling.")
+        if img.shape[1:] != weight_map.shape[1:]:
+            raise ValueError(f"image and weight map spatial shape mismatch: {img.shape[1:]} vs {weight_map.shape[1:]}.")
+        self.randomize(weight_map)
+        _spatial_size = fall_back_tuple(self.spatial_size, weight_map.shape[1:])
+        results = []
+        for center in self.centers:
+            cropper = SpatialCrop(roi_center=center, roi_size=_spatial_size)
+            results.append(cropper(img))
+        return results
```

Comment:
Also here, if `num_samples` is 1 returning `results[0]` may make this more usable with other transforms.